### ISSUE TITLE
label_statistics fails on M1 macs with UNSUPPORTED error

### ISSUE THREAD

**Issue by thawn** [haesleinhuepf/napari-simpleitk-image-processing#17]
trying to run label_statistics on an M1 mac fails with the following openCL error:

```Python
RuntimeError: clBuildProgram failed: BUILD_PROGRAM_FAILURE - clBuildProgram failed: BUILD_PROGRAM_FAILURE - clBuildProgram failed: BUILD_PROGRAM_FAILURE

Build on <pyopencl.Device 'Apple M1 Max' on 'Apple' at 0x6000005f1160>:

Compiler encountered an internal error
(options: -I /opt/homebrew/Caskroom/mambaforge/base/envs/dbnapari-arm64/lib/python3.9/site-packages/pyopencl/cl)
(source saved as /var/folders/8s/7jr95f1d28vflnwm1cfqv7t40000gr/T/tmpr65qvn1f.cl)
```
See the [minimal (non) working example notebook](https://github.com/jqwhite/quail/blob/main/minimal_example.ipynb)

**Comment by haesleinhuepf** [comment]
Oh great, thanks for reporting and thanks for the notebook with the complete error message. That is the key part:

```
File /opt/homebrew/Caskroom/mambaforge/base/envs/dbnapari-arm64/lib/python3.9/site-packages/napari_simpleitk_image_processing/_simpleitk_image_processing.py:837, in label_statistics(intensity_image, label_image, size, intensity, perimeter, shape, position, moments, napari_viewer)
    808 """Measure intensity/shape/... statistics per label
    809 
    810 Parameters
   (...)
    833 ..[1] http://insightsoftwareconsortium.github.io/SimpleITK-Notebooks/Python_html/35_Segmentation_Shape_Analysis.html
    834 """
    835 import SimpleITK as sitk
--> 837 sitk_label_image = sitk.GetImageFromArray(label_image.astype(int))
```

[In this line](https://github.com/haesleinhuepf/napari-simpleitk-image-processing/blob/3dff808160512fefcffd4131f22a4918eaab59a0/src/napari_simpleitk_image_processing/_simpleitk_image_processing.py#L837), a `np.asarray(label_image)` is missing,

Furthermore, we should test if `cl_image.astype(int)` works on M1 GPUs in pyclesperanto.

So both errors appear fixable! 

Thanks again for reporting!

**Comment by haesleinhuepf** [comment]
Hi @thawn ,

this should be fixed in [napari-simpleitk-image-processing 0.4.5](https://github.com/haesleinhuepf/napari-simpleitk-image-processing/releases/tag/0.4.5) 

Please reopen this issue if not!

Thanks for reporting!

Best,
Robert
